source/reference/ejson/parse.txt

```diff
@@ -0,0 +1,161 @@
+
+Behavior
+--------
+
+You can call ``EJSON.parser()`` from inside an interactive ``mongosh``
+session or from the system command line using ``--eval``.
+
+Call ``EJSON.parser()`` from an interactive session: 
+
+.. code-block:: javascript
+   :copyable: false
+
+   EJSON.parse(string)
+
+Call ``EJSON.parser()`` from the system command line: 
+
+.. code-block:: shell
+   :copyable: false
+
+   mongosh --eval "EJSON.parse(string)"
+
+Examples
+--------
+
+To try these examples, first create the ``sales`` collection:
+
+.. code-block:: javascript
+
+   db.sales.insertMany( [ 
+      { custId: 345, purchaseDate: ISODate("2023-07-04"), quantity: 4, cost: Decimal128("100.60"), },
+      { custId: 346, purchaseDate: ISODate("2023-07-12"), quantity: 3, cost: Decimal128("175.45"), },
+      { custId: 486, purchaseDate: ISODate("2023-08-01"), quantity: 9, cost: Decimal128("200.53"), },
+   ] )
+
+Format Input with EJSON.parse()
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+``EJSON.parse()`` accepts a string as input. For this example, use the
+:ref:`EJSON.stringify() <mongosh-ejson-stringify>` method to export the
+``sales`` collection as a string. 
+
+.. code-block:: javascript
+
+   let salesCollection =  EJSON.stringify( db.sales.find().toArray() )   
+
+Use ``EJSON.parse()`` to format the string for use with methods like
+:method:`db.collection.insertMany()` that expect JSON pairs:
+
+.. code-block:: javascript
+
+   db.salesRestored.insertMany( EJSON.parse( salesCollection ) )
+
+- ``EJSON.parse()`` formats the values in ``salesCollection`` as JSON
+  pairs.
+
+- :method:`db.salesRestored.insertMany() <db.collection.insertMany()>`
+  uses the JSON pairs to create the ``salesRestored`` collection.
+
+Use EJSON.parse() from the command line
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To import string data from an external source such as a file or an API
+call, use ``EJSON.parse()`` with the ``mongosh --eval`` method.
+
+For this example, save the ``sales`` collection as a file.
+
+.. code-block:: javascript
+
+   let salesCollection =  EJSON.stringify( db.sales.find().toArray() )   
+
+   fs.writeFileSync( 'sales.json', salesCollection ) 
+
+The code creates a file on your local system called ``sales.json``. To
+import the file and create a new collection, exit ``mongosh`` and run an
+``--eval`` operation from the command line. 
+
+.. code-block:: javascript
+
+   # Note: This example is formatted to fit on the page.
+
+   mongosh --quiet \
+           --eval "db.salesFromFile.insertMany( \
+              EJSON.parse( fs.readFileSync( 'sales.json', 'utf8' ) ) )"
+
+- ``EJSON.parse()`` takes a string as input. This example uses
+  ``fs.readFileSync()`` to read the ``sale.json`` file as a string. 
+
+- ``EJSON.parse()`` formats the input string as JSON pairs.
+
+- ``db.salesFromFile.insertMany()`` creates the ``salesFromFile``
+  collection from the JSON pairs.
+
+Learn More
+----------
+
+`EJSON documentation <https://www.npmjs.com/package/bson#EJSON>`__
```

Comment:
[suggestion]

Consider using replacements or constants here if they fit in. Thanks!

Reply:
Done